pkg/cloud/gcp/gcp.go

```diff
@@ -100,101 +151,59 @@ func (g *gcpProvider) PrepareSubmarinerClusterEnv() error {
+	if err != nil {
+		return err
 	}
-
 	return operatorhelpers.NewMultiLineAggregate(errs)
 }
 
-func (g *gcpProvider) openPorts(ingress, egress *compute.Firewall) error {
-	var errs []error
-
-	for _, rule := range []*compute.Firewall{ingress, egress} {
-		current, err := g.gcpClient.GetFirewallRule(rule.Name)
-		if isNotFound(err) {
-			errs = append(errs, g.gcpClient.InsertFirewallRule(rule))
-			continue
-		}
-
-		if err != nil {
-			errs = append(errs, err)
-			continue
-		}
-
-		if !isChanged(current, rule) {
-			continue
-		}
+func newClient(kubeClient kubernetes.Interface, secretNamespace, secretName string) (string, gcpclient.Interface, error) {
+	credentialsSecret, err := kubeClient.CoreV1().Secrets(secretNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return "", nil, err
+	}
 
-		errs = append(errs, g.gcpClient.UpdateFirewallRule(rule.Name, rule))
+	authJSON, ok := credentialsSecret.Data[gcpCredentialsName]
+	if !ok {
+		return "", nil, fmt.Errorf("the gcp credentials %s is not in secret %s/%s", gcpCredentialsName, secretNamespace, secretName)
 	}
 
-	return operatorhelpers.NewMultiLineAggregate(errs)
-}
+	ctx := context.TODO()
 
-func isNotFound(err error) bool {
-	if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
-		return true
+	// since we're using a single creds var, we should specify all the required scopes when initializing
+	creds, err := google.CredentialsFromJSON(ctx, authJSON, dns.CloudPlatformScope)
+	if err != nil {
+		return "", nil, err
 	}
-	return false
-}
 
-func isChanged(oldRule, newRule *compute.Firewall) bool {
-	if len(oldRule.Allowed) != len(newRule.Allowed) {
-		return true
+	// Create a GCP client with the credentials.
+	computeClient, err := gcpclient.NewClient(creds.ProjectID, []option.ClientOption{option.WithCredentials(creds)})
+	if err != nil {
+		return "", nil, err
 	}
 
-	if oldRule.Allowed[0].IPProtocol != newRule.Allowed[0].IPProtocol {
-		return true
-	}
+	return creds.ProjectID, computeClient, nil
 
-	return !reflect.DeepEqual(oldRule.Allowed[0].Ports, newRule.Allowed[0].Ports)
 }
 
-func newFirewallRules(name, projectID, infraId, protocol string, ports []string) (ingress, egress *compute.Firewall) {
-	ingressName, egressName := generateRuleNames(infraId, name)
-	return &compute.Firewall{
-			Name:      ingressName,
-			Network:   fmt.Sprintf("projects/%s/global/networks/%s-network", projectID, infraId),
-			Direction: "INGRESS",
-			Allowed: []*compute.FirewallAllowed{
-				{
-					IPProtocol: protocol,
-					Ports:      ports,
-				},
-			},
-		},
-		&compute.Firewall{
-			Name:      egressName,
-			Network:   fmt.Sprintf("projects/%s/global/networks/%s-network", projectID, infraId),
-			Direction: "EGRESS",
-			Allowed: []*compute.FirewallAllowed{
-				{
-					IPProtocol: protocol,
-					Ports:      ports,
-				},
-			},
-		}
+func (g *gcpProvider) Started(message string, args ...interface{}) {
+	klog.Infof(fmt.Sprintf(message, args...))
```

Comment:
This should use the event recorder instead (see the AWS PR for an example). The event recorder feeds the logs and the event journal in OCP.